app/src/main/java/com/example/happymeals/database/FireAuth.java

```diff
@@ -0,0 +1,121 @@
+package com.example.happymeals.database;
+
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.example.happymeals.OutputListener;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/*
+
+ */
+
+
+public class FireAuth {
+
+    // create FireAuth to function statically and allow login and register to access it
+    public FirebaseFirestore fireStore;
+    public FirebaseAuth authenticate;
+    static private FireAuth fireauth;
+
+    // getter and setter for FireAuth
+    public static FireAuth getFireauth() {
+        // check if empty than initialize
+        if(fireauth == null) {
+            fireauth = new FireAuth();
+            fireauth.authenticate = FirebaseAuth.getInstance();
```

Comment:
Why is this not being done in the constructor?

Reply:
It would do the same thing, but i'll rewrite it np.